### PR TITLE
Make infinity rerank more compatible with jina

### PR DESCRIPTION
**Pull request by shizidushu**
As described in https://jina.ai/reranker/, jinja rerank

- defaults to return documents (but infinity default to false)
- returned document (reponse ) lives in `text` property.

As current some project (e.g., dify) does not support infinity, let infinity has the same default behavior with jinja is helpful.

API's to consider:
- https://docs.voyageai.com/reference/reranker-api `{return_documents: bool = False} {.., documents: str}`
- https://jina.ai/reranker/ `{return_documents: bool = False} -> {.., documents: str}`
- https://docs.cohere.com/reference/rerank `{no_kwarg_for_documents} -> {document: {text: str}}`
- https://huggingface.github.io/text-embeddings-inference `{return_text: bool = False} -> {.., text: str}`

### CODE REVIEW

File: libs/infinity_emb/infinity_emb/fastapi_schemas/pymodels.py [hunk 1]
```diff
@@ -223,7 +227,11 @@ def to_rerank_response(
             return dict(
                 model=model,
                 results=[
-                    dict(relevance_score=score, index=count, document=doc)
+                    dict(
+                        relevance_score=score,
+                        index=count,
+                        document=_ReRankDocument(text="doc"),
```
[the remark this answers is not in the document]
Comment (shizidushu) [reply libs/infinity_emb/infinity_emb/fastapi_schemas/pymodels.py]: Yes. It is a typo. It should be `document=_ReRankDocument(text=doc)`

**Comment by shizidushu** [comment]
> @shizidushu Do you mean jina?
> 
> The bot's issues above are all three correct points. I think a change does not make sense here, but thank you for your contribution!
> 
> https://docs.cohere.com/reference/rerank and huggingface TEI set default=False for return documents. https://huggingface.github.io/text-embeddings-inference. The reason for it, is to save bandwidth by default.

@michaelfeil 

Yes. It is Jina. Sorry for the typo.

Thanks for you explanation. Yes, it is ok to set default=False for return documents.

But I still suggest to follow the convention of cohere and jina for the data structure of document:

图片

图片

**Comment by michaelfeil** [comment]
Fair point - I interpreted the API from cohere incorrectly. Jina released their rerank api later, afaik. 

Potentially, it would make more sense to align the API spec with Text-embedding-inference. 
https://huggingface.github.io/text-embeddings-inference/#/Text%20Embeddings%20Inference/rerank 

In this case making an alias for return_text and https://docs.pydantic.dev/latest/concepts/alias/#aliaspath-and-aliaschoices
```
return_text: str = Field(validation_alias=AliasChoices('return_text', 'return_documents'))
```

Would that work for your use-case?

**Comment by shizidushu** [comment]
@michaelfeil No. In my case, I have to change both default of return documents and use 'text' property, i.e., align the behaviour with jinja. (Otherwise Dify will report errors). Because I need to pretend it is a jina rank api

图片

**Comment by michaelfeil** [comment]
Dify has also an integration for text-embeddings inference, right? Lets align on TEI's integation.

**Comment by shizidushu** [comment]
@michaelfeil Thanks. I got it.

**Pull request closed without merging**